### PR TITLE
LPS-135803 Add new option to general panel for displaying all collection items

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

**Comment by veroglez** [comment]
ci:test:sf

**Comment by veroglez** [comment]
ci:test:relevant

File: modules/apps/layout/layout-api/src/main/java/com/liferay/layout/util/structure/CollectionStyledLayoutStructureItem.java [hunk 1]
```diff
@@ -133,6 +131,14 @@ public String getPaginationType() {
 		return _paginationType;
 	}
 
+	/**
+	 * @deprecated As of Athanasius (7.3.x), with no direct replacement
+	 */
+	@Deprecated
```
Comment (victorg1991) [inline modules/apps/layout/layout-api/src/main/java/com/liferay/layout/util/structure/CollectionStyledLayoutStructureItem.java:137]: Probably copy paste error? why is this poor new method already deprecated? :'(

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js [hunk 2]
```diff
@@ -117,6 +119,17 @@ export const CollectionGeneralPanel = ({item}) => {
 		}
 	}, [collectionItemType]);
 
+	useEffect(() => {
+		if (showAllItems) {
+			CollectionService.getCollectionItemCount({
+				collection: item.config.collection,
+				onNetworkStatus: () => {},
+			}).then(({totalNumberOfItems}) => {
+				setTotalNumberOfItems(totalNumberOfItems);
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js:128]: I think we should wrap check here that the component is mounted. We can use the hook `useIsMounted`

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js [hunk 3]
```diff
@@ -286,22 +288,47 @@ export const CollectionGeneralPanel = ({item}) => {
...
+										)}
+										onChange={({target}) => {
+											setShowAllItems(target.checked);
+
+											handleConfigurationChanged({
+												showAllItems: target.checked,
+											});
+										}}
+										onDoubleClick={(event) => {
+											event.stopPropagation();
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js:343]: Why is this needed?

**Comment by victorg1991** [comment]
LGTM :)  I've added a couple of comments, but nothing really critical

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js (hunk 3 above)
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js]: Probably another copy paste error, it is already removed.

File: modules/apps/layout/layout-api/src/main/java/com/liferay/layout/util/structure/CollectionStyledLayoutStructureItem.java (hunk 1 above)
Comment (veroglez) [reply modules/apps/layout/layout-api/src/main/java/com/liferay/layout/util/structure/CollectionStyledLayoutStructureItem.java]: Yes, my mistake. It is already solved

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js (hunk 2 above)
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CollectionGeneralPanel.js]: I did not know this. Thanks for the info! 👏

**Comment by veroglez** [comment]
ci:test:sf

**Comment by veroglez** [comment]
ci:test:relevant

**Comment by victorg1991** [comment]
ci:forward
